controllers/topology/cluster_controller_test.go

```diff
@@ -364,13 +364,13 @@ func setupTestEnvForIntegrationTests(ns *corev1.Namespace) (func() error, error)
 	// The objects are created for every test, though some e.g. infrastructureMachineTemplate2 may not be used in every test.
 	initObjs := []client.Object{
 		clusterClass,
-		cluster1,
-		cluster2,
 		infrastructureClusterTemplate,
 		infrastructureMachineTemplate,
 		infrastructureMachineTemplate2,
 		bootstrapTemplate,
 		controlPlaneTemplate,
+		cluster1,
```

Comment:
Any guesses for why I needed to move the clusters down the list here? 😆

Reply:
I couldn't reproduce any errors, but I assume you got an error that the ClusterClass doesn't exist when cluster1 was created?

I think the problem is that local cache of the `mgr.GetClient` doesn't have the ClusterClass yet. Moving clusters a bit down gives higher probability to avoid the race condition. (but I think it doesn't fix it 100%)

If that's the case here I wonder if we should use a live client in the webhook in our tests to avoid the accepted and expected issue of a slightly stale cache in our tests.

Reply:
Let's leave it as is for now but reserve this as a solution if we see problems. These webhooks are going to have to make a number of API calls so the impact of changing to a live client could be high.

Reply:
Yeah let's see. But let's try to keep it in mind if we see a bunch of tests which are using ClusterClass are getting flaky in CI.

Reply:
It might be useful to have a CreateAndWait or something similar in our `env` APIs to wait for an object to show up in a Get() call with a ResourceVersion

Reply:
Looks like there's already a CreateAndWait implemented in our internal env package 😄. I've updated the usage here. https://github.com/kubernetes-sigs/cluster-api/blob/e09ed61cc9ba8bd37b0760291c833b4da744a985/internal/envtest/environment.go#L364